generator.go

```diff
@@ -66,27 +71,45 @@ func NewV5(ns UUID, name string) UUID {
 	return DefaultGenerator.NewV5(ns, name)
 }
 
-// NewV6 returns a k-sortable UUID based on a timestamp and 48 bits of
+// NewV6 returns a k-sortable UUID based on the current timestamp and 48 bits of
 // pseudorandom data. The timestamp in a V6 UUID is the same as V1, with the bit
 // order being adjusted to allow the UUID to be k-sortable.
 func NewV6() (UUID, error) {
 	return DefaultGenerator.NewV6()
 }
 
-// NewV7 returns a k-sortable UUID based on the current millisecond precision
-// UNIX epoch and 74 bits of pseudorandom data. It supports single-node batch generation (multiple UUIDs in the same timestamp) with a Monotonic Random counter.
+// NewV6 returns a k-sortable UUID based on the provided timestamp and 48 bits of
+// pseudorandom data. The timestamp in a V6 UUID is the same as V1, with the bit
+// order being adjusted to allow the UUID to be k-sortable.
+func NewV6AtTime(atTime time.Time) (UUID, error) {
+	return DefaultGenerator.NewV6AtTime(atTime)
+}
+
+// NewV7 returns a k-sortable UUID based on the current millisecond-precision
+// UNIX epoch and 74 bits of pseudorandom data. It supports single-node batch
+// generation (multiple UUIDs in the same timestamp) with a Monotonic Random counter.
 func NewV7() (UUID, error) {
 	return DefaultGenerator.NewV7()
 }
 
+// NewV7 returns a k-sortable UUID based on the provided millisecond-precision
+// UNIX epoch and 74 bits of pseudorandom data. It supports single-node batch
+// generation (multiple UUIDs in the same timestamp) with a Monotonic Random counter.
+func NewV7AtTime(atTime time.Time) (UUID, error) {
+	return DefaultGenerator.NewV7AtTime(atTime)
+}
+
 // Generator provides an interface for generating UUIDs.
 type Generator interface {
 	NewV1() (UUID, error)
+	NewV1AtTime(time.Time) (UUID, error)
```

Comment:
I think it is fine to remove from the interface, but that will mean we cannot use the package-level convenience methods, because those operate using `DefaultGenerator` which uses this interface.

This just means that to use the new methods users will need to create a custom generator.

Reply:
I'm not against adding the methods to the interface, we just need to decide whether we bump to v6.x or we accept what is technically a breaking change in a v5.x minor release and "fix with words" for any consumers that have their own implementations.

Reply:
Easy enough for this code, but It's a much bigger deal for every consumer to have to update every import to `s|v5|v6` 😬

Reply:
Unfortunately, there's no way for us to know how many consumers are implementing the interface separately. Numbers like that would be very useful to this decision

Reply:
per pkg.go.dev, 749 public things import /v5.  I know we have a bunch of internal modules at work that also import it.

I can confirm that we don't have any custom `Generator` implementations and someone could theoretically check all of those public importers, but it's impossible to know if any non-public consumers will break.

Personally, I feel like the odds are very low that this will *actually* be a breaking change and we should do a v5.x minor release with a note.

Reply:
Sounds good to me, but we probably should wait for at least one additional voice.